client/js/lounge.js

```diff
@@ -442,15 +453,49 @@ $(function() {
 	});
 
 	contextMenu.on("click", ".context-menu-item", function() {
-		switch ($(this).data("action")) {
+		const $this = $(this);
+		const thisData = $this.data("data");
+		let chan;
+		switch ($this.data("action")) {
 		case "close":
```

Comment:
I feel like we should be starting to use the command pattern with these actions now rather than just increasing the case statement over time.

Reply:
Can you elaborate, not sure what you mean by this.

Reply:
I don't really know what to say to elaborate.

Instead of having a case statement going through all of these actions, which can get very messy and long, we should have a command for each of them that are registered with a click action. These can live in separate places. If you don't know what the command pattern is, go [read up on it](https://en.wikipedia.org/wiki/Command_pattern). It's a very useful design pattern to know.

Reply:
Ah, gotcha. Was indeed not familiar with it.  So something like this then?

```Javascript
var contextActions = {
    close: function(thisData) {
        $(`.networks .chan[data-target="${thisData}"] .close`).click();
    },
    chan: function(thisData) {
        $(`.networks .chan[data-target="${thisData}"]`).click();
    },
    whois: function(thisData) {
        const chan = utils.findCurrentNetworkChan(thisData);

        if (chan.length) {
            chan.click();
        }

        socket.emit("input", {
            target: $("#chat").data("id"),
            text: "/whois " + thisData,
        });

        $(`.channel.active .users .user[data-name="${thisData}"]`).click();
    },
    query: function(thisData) {
        const chan = utils.findCurrentNetworkChan(thisData);

        if (chan.length) {
            chan.click();
        }

        socket.emit("input", {
            target: $("#chat").data("id"),
            text: "/query " + thisData,
        });
    },
    kick: function(thisData) {
        socket.emit("input", {
            target: $("#chat").data("id"),
            text: "/kick " + thisData,
        });
    }
};

contextActions.execute = function(name) {
    return contextActions[name] && contextActions[name].apply(contextActions, [].slice.call(arguments, 1));
};
```
Somewhere tucked away. And then instead of the switch statement do something like

```Javascript

contextMenu.on("click", ".context-menu-item", function() {
    const $this = $(this);
    const thisData = $this.data("data");
    const contextAction = $this.data("action");
    contextActions.execute(contextAction, thisData)
});
```

Reply:
Almost, yeah. That would be a good start (probably good enough  for this PR) but at some point each of the contextActions should be in a separate file (1 for kick, 1 for query, etc). But as there aren't a huge number of them, probably not necessary yet, but wouldb egood at some point.